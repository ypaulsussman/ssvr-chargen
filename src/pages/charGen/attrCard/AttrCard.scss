.attrcard {
  padding: 10px 40px;
  margin-top: 20px;
  margin-bottom: 20px;
  border-radius: 6px;
  background-color: #fcf0ee;
  box-shadow: 6px 6px 18px 2px rgba(0, 0, 0, 0.75);
  text-align: left;
  &--pinned {
    box-shadow: inset 0px 0px 18px 4px rgba(0, 0, 0, 0.75);
  }

  &--dependent {
    margin-left: 60px;
  }

  &__name,
  &__description {
    cursor: default;
    white-space: pre-line;
  }
}
