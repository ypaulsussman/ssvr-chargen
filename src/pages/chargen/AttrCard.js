import React from "react";

const AttrCard = ({ attr, togglePin, pinKeys, unPinKeys }) => {
  const togglePinKeys = attr.pinned ? unPinKeys : pinKeys;
  const toggleVal = attr.pinned ? false : true;

  return (
    <div onClick={() => togglePin(togglePinKeys, toggleVal)}>
      <h4>{attr.name}</h4>
      <p>{attr.description}</p>
    </div>
  );
};

export default AttrCard;
