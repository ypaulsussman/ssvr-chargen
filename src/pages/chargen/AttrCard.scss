.attrcard{
  padding: 10px 20px;
  margin-top: 20px;
  margin-bottom: 20px;
  border-radius: 6px;
  background-color: #FCF0EE;
  box-shadow: 0px 0px 12px 0px rgba(0,0,0,0.75);
  
  &--pinned {
    box-shadow: inset 0px 0px 12px 0px rgba(0,0,0,0.75);
    // background-color: salmon;
  }
  
  &--dependent {
    margin-left: 40px;
  }

  &__name, &__description {
    cursor: default;
  }

  &__description {
    letter-spacing: .25px;
  }
}
