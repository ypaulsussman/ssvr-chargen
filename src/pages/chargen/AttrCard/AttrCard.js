import React from "react";
import "./AttrCard.scss";

const AttrCard = ({ attr, headTitle, togglePin, pinKeys, unPinKeys }) => {
  const togglePinKeys = attr.pinned ? unPinKeys : pinKeys;
  const toggleVal = attr.pinned ? false : true;
  const toggle = () => togglePin(togglePinKeys, toggleVal);
  const cardClassNames = `attrcard 
  attrcard--${attr.pinned ? "pinned" : "unpinned"}
  attrcard--${pinKeys.length > 1 ? "dependent" : "independent"}
  `;

  return (
    <div
      onClick={toggle}
      className={cardClassNames}
      role="button"
      title="Click to pin this attribute (and any parents) for future randomizations"
    >
      <h3 onClick={toggle} className="attrcard__name">
        {headTitle}: {attr.name}
      </h3>
      <p onClick={toggle} className="attrcard__description">{attr.description}</p>
    </div>
  );
};

export default AttrCard;
