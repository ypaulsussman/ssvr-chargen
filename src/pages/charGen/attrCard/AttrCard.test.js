import React from "react";
import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import AttrCard from "./AttrCard";

describe("PC-Attribute Cards", () => {
  let attr, headTitle, togglePin, pinKeys, unPinKeys;

  beforeEach(() => {
    attr = {
      name: "foo",
      description: "bar baz",
      pinned: true,
    };
    headTitle = "meep morp";
    togglePin = jest.fn();
    pinKeys = ["quuz"];
    unPinKeys = ["quuz"];
  });

  afterEach(() => {
    cleanup;
  });

  test("should render", () => {
    render(
      <AttrCard
        attr={attr}
        headTitle={headTitle}
        togglePin={togglePin}
        pinKeys={pinKeys}
        unPinKeys={unPinKeys}
      />
    );
    expect(document.body).toMatchSnapshot();
  });

  test("should call the toggle-fn when clicked", () => {
    render(
      <AttrCard
        attr={attr}
        headTitle={headTitle}
        togglePin={togglePin}
        pinKeys={pinKeys}
        unPinKeys={unPinKeys}
      />
    );
    fireEvent.click(screen.getByTitle(/Click to pin this attribute/))
    expect(togglePin).toHaveBeenCalledWith(pinKeys, false)
  });
});
