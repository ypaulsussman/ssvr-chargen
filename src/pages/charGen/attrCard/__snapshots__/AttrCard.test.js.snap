// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PC-Attribute Cards should render 1`] = `
<body>
  <div>
    <button
      class="attrcard 
  attrcard--pinned
  attrcard--independent
  "
      title="Click to pin this attribute (and any parents) for future randomizations"
    >
      <h3
        class="attrcard__name"
      >
        meep morp
        : 
        foo
      </h3>
      <p
        class="attrcard__description"
      >
        bar baz
      </p>
    </button>
  </div>
</body>
`;
